### ISSUE TITLE
.Equals() does not work?

### ISSUE THREAD

**Issue by jeremy-farrance** [2sic/2sxc#3213]
### ...about
[x] Razor templating
[x] other / unknown

### Current Behavior
.Child().Equals() does not match as expected in this scenario. No error is thrown either.

```
@inherits Custom.Hybrid.RazorTyped

@{
  var activePersons = AsItems(App.Data["Person"])
    .Where(p => p.Bool("Active"))
  ;

  @* 1. if the URL params provide a person ID, get that Person *@
  var currPerson = activePersons 
    .Where(p => p.Id == MyPage.Parameters.Int("person"))
    .FirstOrDefault()
  ;
  if ( currPerson == null ) {
    currPerson = activePersons.FirstOrDefault();
  }

  @* 2. for the selected person, get their list of assigned badges *@
  var bas = AsItems(App.Data["BadgeAssignments"]);
    bas.Where(ba => ba.Child("Person").Equals(currPerson))   // <<== here
    .Select(b => b.Child("Badge"))
  ;
}

<div class="mb-3">
  @foreach ( var badge in badges )
  {
    <span>
      @badge.Picture("Image", 
        settings: "Square",
        width: 54,
        imgAlt: badge.String("Name"))
    </span>  
  }
</div>
```

### Expected Behavior
It seems like the result of .Child() and currPerson should be the same .GetType() and also match using .Equals(), but they do not. I debugged a bit by logging 

`ba.Child("Person").GetType()`:  ToSic.Sxc.Data.TypedItem (as expected) 
`currPerson.GetType()`: ToSic.Sxc.Data.TypedItemOfEntity (not expected?)

### Instructions to Reproduce the Problem
In addition to the code above, if needed to reproduce the scenario, there is more detail [here](https://dnncommunity.org/blogs/Post/17235/Related-Data-in-2sxc-Part-2) and [here](https://dnncommunity.org/blogs/Post/17212/Related-Data-in-2sxc-with-RazorTyped). In Part 2, search for "This does not work either" to see the context discussed.
 
I also tried wrapping the result in AsItem(), but there was still no match.

### Your environment

* **2sxc version(s):** 16.07.00
* **Browser:** all/any 
* **DNN:** 9.11.02 || 9.13.00
* **Hosting platform:** Windows Server 2019 / IIS
* **Language:** any/all

**Comment by jeremy-farrance** [comment]
Also, this was discussed [here (Discord thread)](https://discord.com/channels/706654343176126496/1173523319241310268) in some detail also.

**Comment by iJungleboy** [comment]
Could reproduce and found the cause. 

Internally we have a concept of wrapping objects to extend the information they carry - eg. that an entity which was a child of something knows about that (eg. to behave differently when a toolbar is shown for a child, which is in a list, vs. a parent).

The ITypedItem was only working with IWrapper, not IMultiWrapper - so the inner equality hit the first wrapper (containing list information) and not the root object. 

Fixed in 17.00.01
